### PR TITLE
Only return unique itineraries.

### LINKED ISSUE #256: Same route shown multiple times
image
The same route was just repeated 3 times as my options. There should only be one option displayed if there is only one reasonable route.

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Fixes #256.
This is a workaround for opentripplanner/OpenTripPlanner#1894.

### CODE REVIEW

File: src/app/scripts/cac/control/cac-control-itinerary-list.js [hunk 1]
```diff
@@ -52,7 +52,21 @@ CAC.Control.ItineraryList = (function ($, Handlebars, Utils) {
...
+        // Due to issue: https://github.com/opentripplanner/OpenTripPlanner/issues/1894
+        // itineraries with transit + (bike/walk) can return 3 identical itineraries if only
+        // bike/walk used, and not transit.
+        itineraries = [];
+        if (newItineraries.length > 0) {
+            // _.findWhere returns null if list is empty, so prime list with first itinerary
+            itineraries.push(newItineraries[0]);
+        }
+        _.each(newItineraries, function(itinerary) {
+            if (_.findWhere(itineraries, itinerary) === null) {
```
Comment (kshepard) [inline src/app/scripts/cac/control/cac-control-itinerary-list.js]: This could probably be:
`if (!_.findWhere(itineraries, itinerary))`

**Comment by kshepard** [comment]
+1

File: src/app/scripts/cac/control/cac-control-itinerary-list.js (hunk 1 above)
Comment (flibbertigibbet) [inline src/app/scripts/cac/control/cac-control-itinerary-list.js]: Yeah, I guess it's never going to evaluate to 0. That looks better.

Comment (flibbertigibbet) [inline src/app/scripts/cac/control/cac-control-itinerary-list.js]: Interestingly, that didn't seem to work. I'll just leave it as-is.

**Pull request merged**